src/plugins/intel_cpu/src/nodes/mkldnn_shuffle_channels_node.cpp

```diff
@@ -123,7 +160,20 @@ void MKLDNNShuffleChannelsNode::createPrimitive() {
 
 void MKLDNNShuffleChannelsNode::prepareParams() {
     auto& srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
-    execPtr = std::make_shared<ShuffleChannelsExecutor>(attrs, srcMemPtr->getStaticDims(), srcMemPtr->GetDescWithType<BlockedMemoryDesc>()->getBlockDims());
+    auto builder = [](const ShuffleChannelsKey& key) -> std::shared_ptr<ShuffleChannelsExecutor> {
+        return std::make_shared<ShuffleChannelsExecutor>(key.attrs, key.srcDims, key.srcBlockedDims);
+    };
+
+    ShuffleChannelsKey key = {attrs,
```

Comment:
As I can see, the `attrs` part of the key does not change, only `srcDims` and `srcBlockedDims` are changed. But every time we copy this `attrs` part. What if we store the key in the node instead of attrs, so we could reuse the most part of the key changing only dimensions. Or, we could extend `attrs` with the dimensions vectors and add `hash` and `operator==` so the `attrs` could be used as a key. The same is applicable to the `SpaceToDepth` node. What do you think?

Reply:
Agreed! will extend the attr in the next version update .